src/coreComponents/mesh/utilities/MeshMapUtilities.hpp

```diff
@@ -225,6 +225,88 @@ inline localIndex numElements( ArrayOfSetsView< T const > const & map )
   return size;
 }
 
+/**
+ * @brief Transposes an input map (array2d, ArrayOfArrays of ArrayOfSets)
+ * @tparam POLICY execution policy to use
+ * @tparam VIEW_TYPE type of view of the source map
+ * @param srcSize number of source objects ("rows" of @p srcMap)
+ * @param dstSize number of target objects ("cols" of @p srcMap)
+ * @param srcMap the source map
+ * @param overAlloc overallocation (extra capacity per row) for resulting map
+ * @return the transpose of @p srcMap stored as ArrayOfArrays (most general form)
+ */
+template< typename POLICY, typename VIEW_TYPE >
+ArrayOfArrays< std::remove_const_t< typename VIEW_TYPE::ValueType > >
+transposeMeshMap( localIndex const srcSize,
+                  localIndex const dstSize,
+                  VIEW_TYPE const & srcMap,
+                  localIndex const overAlloc = 0 )
+{
+  // Count the number of elements in each set
+  array1d< localIndex > counts( dstSize );
+  counts.setValues< POLICY >( overAlloc );
+  forAll< POLICY >( srcSize, [srcMap, counts = counts.toView()] ( localIndex const srcID )
+  {
+    for( localIndex const dstID : srcMap[ srcID ] )
+    {
+      RAJA::atomicInc< AtomicPolicy< POLICY > >( &counts[ dstID ] );
+    }
+  } );
+
+  // Allocate storage for the transpose map
+  ArrayOfArrays< localIndex > dstMap;
+  dstMap.resizeFromCapacities< parallelHostPolicy >( dstSize, counts.data() );
+
+  // Fill the sub-arrays with unsorted entries
+  forAll< POLICY >( srcSize, [srcMap, dstMap = dstMap.toView()] ( localIndex const srcID )
+  {
+    for( localIndex const dstID : srcMap[ srcID ] )
+    {
+      dstMap.emplaceBackAtomic< AtomicPolicy< POLICY > >( dstID, srcID );
+    }
+  } );
+
+  return dstMap;
+}
+
+/**
+ * @brief Turn ArrayOfArrays into an ArrayOfSets
+ * @tparam POLICY execution policy to use for sorting entries
+ * @tparam T value type
+ * @tparam COMP type comparison function, defaults to less
+ * @param input the source, passed by rvalue ref so its memory is pilfered
+ * @param valuesAreUnique whether values in input are already unique and do not need deduplication
+ * @param comp comparison function, must be copyable and compatible with @p POLICY
+ * @return the resulting ArrayOfSets
+ */
+template< typename POLICY, typename T, typename COMP = LvArray::sortedArrayManipulation::less< T > >
+ArrayOfSets< T >
+makeArrayOfSets( ArrayOfArrays< T > && input, bool valuesAreUnique, COMP comp = {} )
```

Comment:
The only reason this function exists is because `LvArray::ArrayOfSets::assimilate()` does sub-array sorting in serial, but there is a `TODO` comment saying it should take an execution policy argument. I was trying to avoid making an `LvArray` PR, but if it's the right thing to do, I will go ahead and just update `assimilate()`.

Reply:
Well...I am not sure I would say it is the "right way", but it does make `assimilate()` better, and removes this function...so maybe there are advantages to that.